index/scorch/snapshot_index.go

```diff
@@ -401,24 +404,35 @@ func (i *IndexSnapshot) DocumentVisitFieldTerms(id index.IndexInternalID,
 
 	ss := i.segment[segmentIndex]
 
-	err = ss.cachedDocs.prepareFields(fields, ss)
-	if err != nil {
-		return err
-	}
+	switch seg := ss.segment.(type) {
+	case *mem.Segment:
```

Comment:
instead of doing type assertion matching the implementation structure, should we just be checking to see if it supports UnInvertIndex.  If it does, we use that, if it doesn't we fallback to the existing mechanism (which should work for any segment regardless of implementation)